File: 0005-longest-palindromic-substring/0005-longest-palindromic-substring.cpp
class Solution {
public:
    string longestPalindrome(string s) {
        int n=s.size();
        if(n==1)return s;
        int maxL=0,st=0,end=0;
        
        for(int i=0;i<n;i++){
            int l=i,r=i;
            while(l>=0 && r<n && s[l]==s[r]){
                if(maxL<r-l+1){
                    maxL=r-l+1;
                    st=l,end=r;
                }
                l--;
                r++;
            }
        }
        
            for(int i=0;i<n;i++){
            int l=i,r=i+1;
            while(l>=0 && r<n && s[l]==s[r]){
                if(maxL<r-l+1){
                    maxL=r-l+1;
                    st=l,end=r;
                }
                l--;
                r++;
            }
        }
        
        if(maxL==0)maxL=1;
    return s.substr(st,maxL);
        
    }
};

File: 0005-longest-palindromic-substring/0005-longest-palindromic-substring.js
/**
 * @param {string} s
 * @return {string}
 */
var longestPalindrome = function(s) {
    const n=s.length;
    let ans=0,idx=0;
    for(let i=0;i<n;i++){
        let left=i,right=i;
        while(left>=0 && right<n && s[left]==s[right]){
            const curr=right-left+1;
            if(curr>ans){
                ans=curr;
                idx=left;
            }
            
            left--;
            right++;
        }
    }
    for(let i=0;i<n;i++){
        let left=i,right=i+1;
        while(left>=0 && right<n && s[left]==s[right]){
            const curr=right-left+1;
            if(curr>ans){
                ans=curr;
                idx=left;
            }
            left--;
            right++;
        }
    }
    return s.substr(idx,ans);

};